src/stability_sdk/api.py

```diff
@@ -192,8 +632,44 @@ def update(self, title:str=None, file_id:str=None, file_uri:str=None):
             self._project.file.uri = file_uri
 
 
+    @staticmethod
+    def add_asset_metadata(project_id: str, asset_id: str, mime_type: str, filename: str, project_key: str = None) -> None:
+        # metadata_index = self.load_metadata_index() # I assume metadata is updated by each operation
+        if project_id not in Project._metadata_index:
+            Project._metadata_index[project_id] = {}
+        Project._metadata_index[project_id][asset_id] = {
+            "mime_type": mime_type
+        }
+        if filename is not None:
+            Project._metadata_index[project_id][asset_id]["filename"] = filename
+        if project_key is not None:
+            Project._metadata_index[project_id][project_key] = asset_id
+        Project.save_metadata_index()
+
+    @staticmethod
+    def delete_project_metadata(project_id: str) -> None:
+        Project._metadata_index.pop(project_id, None)
+
+    @classmethod
+    def save_metadata_index(cls, metadata_index: dict = None) -> None:
+        if metadata_index is None:
+            metadata_index = cls._metadata_index
+        index_file = f"metadata_index.json"
+        with open(index_file, "w") as f:
+            json.dump(metadata_index, f)
+
+    @classmethod
+    def load_metadata_index(cls) -> dict:
+        index_file = "metadata_index.json"
+        if os.path.exists(index_file):
+            with open(index_file, "r") as f:
+                metadata_index = json.load(f)
+            return metadata_index
+        return {}
```

Comment:
My thinking on this was basically that for future purposes, I have an interest in representing this information in application memory. This metadata JSON saved to disk is an easy dump for bootstrapping reasons since we don't have a UI to look at what the in-memory representation.
I agree that it should currently be considered a cache at best, since someone could potentially create/delete/etc. projects+assets by other means (, and the metadata isn't saying much else interesting aside from what can be found in projects+assets).. and so the sync/refresh strategy needs some work. I've planned to at least add a script to download all assets and rebuild metadata, but haven't written it yet.
For persistent storage, I'd like to see everything kept in the project+assets rather than solely metadata.